modules/gitops-argo-cd-round-robin-sharding-algorithm.adoc

```diff
@@ -0,0 +1,235 @@
+
+.. Observe the number of created application controller pods. It should correspond with the number of set replicas.
+
+.. Click on the controller pod you want to examine and go to the *Logs* tab to view the pod logs.
++
+.Example controller pod logs snippet
+[source,terminal]
+----
+time="2023-12-13T09:05:34Z" level=info msg="ArgoCD Application Controller is starting" built="2023-12-01T19:21:49Z" commit=a3vd5c3df52943a6fff6c0rg181fth3248976299 namespace=openshift-gitops version=v2.9.2+c5ea5c4
+time="2023-12-13T09:05:34Z" level=info msg="Processing clusters from shard 1"
+time="2023-12-13T09:05:34Z" level=info msg="Using filter function:  round-robin" <1>
+time="2023-12-13T09:05:34Z" level=info msg="Using filter function:  round-robin"
+time="2023-12-13T09:05:34Z" level=info msg="appResyncPeriod=3m0s, appHardResyncPeriod=0s"
+----
+<1> Look for the `"Using filter function:  round-robin"` message.
+
+.. In the log *Search* field, search for `processed by shard` to verify that the cluster distribution across shards is even.
++
+[IMPORTANT]
+====
+Ensure that you set the log level to `debug` to observe these logs.
+====
++
+.Example controller pod logs snippet
+[source,terminal]
+----
+time="2023-12-13T09:05:34Z" level=debug msg="ClustersList has 3 items"
+time="2023-12-13T09:05:34Z" level=debug msg="Adding cluster with id= and name=in-cluster to cluster's map"
+time="2023-12-13T09:05:34Z" level=debug msg="Adding cluster with id=068d8b26-6rhi-4w23-jrf6-wjjfyw833n23 and name=in-cluster2 to cluster's map"
+time="2023-12-13T09:05:34Z" level=debug msg="Adding cluster with id=836d8b53-96k4-f68r-8wq0-sh72j22kl90w and name=in-cluster3 to cluster's map"
+time="2023-12-13T09:05:34Z" level=debug msg="Cluster with id= will be processed by shard 0" <1>
+time="2023-12-13T09:05:34Z" level=debug msg="Cluster with id=068d8b26-6rhi-4w23-jrf6-wjjfyw833n23 will be processed by shard 1" <1>
+time="2023-12-13T09:05:34Z" level=debug msg="Cluster with id=836d8b53-96k4-f68r-8wq0-sh72j22kl90w will be processed by shard 2" <1>
+----
+<1> In this example, 3 clusters are attached consecutively to shard 0, shard 1, and shard 2.
++
+[NOTE]
+====
+If the number of clusters "C" is a multiple of the number of shard replicas "R", then each shard must have the same number of assigned clusters "N", which is equal to "C" divided by "R". The example above shows 3 clusters and 3 replicas, therefore each shard has 1 cluster assigned.
+====
+
+[id="enabling-the-round-robin-sharding-algorithm-using-cli_{context}"]
+== Enabling the round-robin sharding algorithm using CLI
+
+You can enable the `round-robin` sharding algorithm using the command-line interface.
+
+.Prerequisites
+* You have installed the {gitops-title} Operator in your cluster.
+* You have access to the cluster with `cluster-admin` privileges.
+
+.Procedure
+
+. Enable sharding and set the number of replicas to the wanted value:
```

Comment:
```suggestion
. Enable sharding and set the number of replicas to the wanted value by running the following command:
```
See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

This would apply to the following steps as well.

Reply:
@ekristova +1 to that comment

Reply:
I prefer no introductions as well. I think someone told me that this makes it easier for localization, but I don't know....

Reply:
I see, maybe that is why... I am meeting for a writer exercise next week, so I will bring this up and maybe learn something about it, if I do, I'll let you know :) Thank you!